Neos.Flow/Classes/ObjectManagement/DependencyInjection/ProxyClassBuilder.php

```diff
@@ -154,7 +154,7 @@ public function build(): void
             $constructor->addPostParentCallCode($this->buildLifecycleInitializationCode($objectConfiguration, ObjectManagerInterface::INITIALIZATIONCAUSE_CREATED));
             $constructor->addPostParentCallCode($this->buildLifecycleShutdownCode($objectConfiguration, ObjectManagerInterface::INITIALIZATIONCAUSE_CREATED));
 
-            if ($this->objectManager->getContext()->isProduction()) {
+            if ($this->objectManager->getContext()->isProduction() || $this->objectManager->getContext()->isTesting()) {
```

Comment:
Interesting question indeed, the compile static always had different behavior in production and development, I wonder though why I had to add this for testing, I'll investigate. The point of static compile Is to bake in expensive operations that cannot change during runtime (eg. all inerface implementations). It did work fine until the recent changes in the framework. this general switch is years old.

Reply:
> this general switch is years old

I'm not sure how it's related. All I can say is that the regression seems to be gone without that check

Reply:
yep, understood! Maybe it was unintended 🤷 I'll backtrace

Reply:
@kitsunet @mhsdesign @bwaidelich Do we still consider this as an issue? Maybe we should create a dedicated ticket then.

For some reason, it's not an issue in "my" behat tests anymore, but I don't know why. And we don't know, why it's not an issue at all in the flow and neos tests - but is easily reproducible with Bastians "CommandController".

Reply:
Maybe we should roll this back together with the test and see if we can come up with a way to enable it for one test, recompile do test and flush/reset afterwards. It might just make sense to introduce this as a setting, that way it would be easier to set it on a case by case basis and people could disable it if it causes issues.